includes/subscriptions/class-wc-payments-product-service.php

```diff
@@ -478,10 +478,78 @@ public function unarchive_price( string $wcpay_price_id, $test_mode = null ) {
 		$this->payments_api_client->update_price( $wcpay_price_id, $data );
 	}
 
+	/**
+	 * Prevents the subscription interval to be greater than 1 for yearly subscriptions.
+	 *
+	 * @param int $product_id Post ID of the product.
+	 */
+	public function limit_yearly_subscription_intervals_on_save_product( $product_id ) {
+		// Skip products that aren't subscriptions.
+		$product = wc_get_product( $product_id );
+		if (
+			! $product ||
+			! WC_Subscriptions_Product::is_subscription( $product ) ||
+			empty( $_POST['_wcsnonce'] ) ||
+			! wp_verify_nonce( sanitize_key( $_POST['_wcsnonce'] ), 'wcs_subscription_meta' )
+		) {
+			return;
+		}
+
+		if (
+			isset( $_REQUEST['_subscription_period'] ) &&
+			'year' === $_REQUEST['_subscription_period'] &&
+			isset( $_REQUEST['_subscription_period_interval'] ) &&
+			'1' !== $_REQUEST['_subscription_period_interval']
+		) {
+			// Prevent WC Subs Core from saving the interval when it's invalid.
+			$_REQUEST['_subscription_period_interval'] = '1';
```

Comment:
@mattallan I know you suggested unsetting this from $_REQUEST / $_POST [here](https://github.com/Automattic/woocommerce-subscriptions-core/pull/30). I opted for forcing it to "1" instead because if the previous value was "every 3rd week", unsetting the value would still create an invalid subscription product.